network-store-server/src/main/java/com/powsybl/network/store/server/LimitsHandler.java

```diff
@@ -0,0 +1,361 @@
+                    selectedOperationalLimitsGroupIdentifiers.operationalLimitsGroupId2(), 2, selectedLimitsInfos);
+                if (operationalLimitsGroupAttributes != null) {
+                    currentLimitsGroupAttributes.computeIfAbsent(owner.getEquipmentId(), k -> new HashMap<>());
+                    currentLimitsGroupAttributes.get(owner.getEquipmentId()).put(OperationalLimitsGroupIdentifier.of(owner.getEquipmentId(),
+                            selectedOperationalLimitsGroupIdentifiers.operationalLimitsGroupId2(), 2),
+                        getSelectedCurrentLimitOperationalLimitsGroupAttributes(
+                            selectedOperationalLimitsGroupIdentifiers.operationalLimitsGroupId1(), 2, selectedLimitsInfos));
+                }
+            }
+        });
+        return currentLimitsGroupAttributes;
+    }
+
+    private OperationalLimitsGroupAttributes getSelectedCurrentLimitOperationalLimitsGroupAttributes(String selectedGroupId, Integer side, LimitsInfos limitsInfos) {
+        List<PermanentLimitAttributes> permanentLimits = limitsInfos.getPermanentLimits().stream().filter(permanentLimitAttributes ->
+                permanentLimitAttributes.getLimitType().equals(LimitType.CURRENT) &&
+                    permanentLimitAttributes.getOperationalLimitsGroupId().equals(selectedGroupId) &&
+                    permanentLimitAttributes.getSide().equals(side))
+            .toList();
+        if (permanentLimits.size() > 1) {
+            throw new PowsyblException("found more than one permanent current limit for group : " + selectedGroupId + " and side : " + side);
+        } else if (permanentLimits.isEmpty()) {
+            return null;
+        }
+        TreeMap<Integer, TemporaryLimitAttributes> temporaryLimitsTreeMap = new TreeMap<>();
+        limitsInfos.getTemporaryLimits().stream().filter(temporaryLimitAttributes ->
+                temporaryLimitAttributes.getLimitType().equals(LimitType.CURRENT) &&
+                    temporaryLimitAttributes.getOperationalLimitsGroupId().equals(selectedGroupId) &&
+                    temporaryLimitAttributes.getSide().equals(side))
+            .forEach(temporaryLimitAttributes -> temporaryLimitsTreeMap.put(temporaryLimitAttributes.getAcceptableDuration(), temporaryLimitAttributes));
+        return OperationalLimitsGroupAttributes.builder()
+            .id(selectedGroupId)
+            .currentLimits(LimitsAttributes.builder()
+                .operationalLimitsGroupId(selectedGroupId)
+                .permanentLimit(permanentLimits.getFirst().getValue())
+                .temporaryLimits(temporaryLimitsTreeMap)
+                .build())
+            .build();
+    }
+
+    private Map<OwnerInfo, SelectedOperationalLimitsGroupIdentifiers> getPartialVariantSelectedOperationalLimitsGroupIds(UUID networkId, int variantNum, int fullVariantNum, ResourceType type, Set<String> tombstonedElements) {
```

Comment:
@antoinebhs could check this part ? it concerns partial variant and i'm not sure

Reply:
it is not limits data but select operational limits group **ID**